From stable-bounces@linux.kernel.org Thu Sep 20 12:56:12 2007
From: Denis V. Lunev <den@openvz.org>
Date: Thu, 20 Sep 2007 12:55:47 -0700 (PDT)
Subject: [stable] [PATCH]: Fix IPV6 DAD handling
To: stable@kernel.org
Cc: bunk@kernel.org
Message-ID: <20070920.125547.35800190.davem@davemloft.net>

From: Denis V. Lunev <den@openvz.org>

commit 9e3be4b34364a670bd6e57d2e8c3caabdd8d89f8 in mainline

addrconf_dad_failure calls addrconf_dad_stop which takes referenced address
and drops the count. So, in6_ifa_put perrformed at out: is extra. This
results in message: "Freeing alive inet6 address" and not released dst entries.

Signed-off-by: Denis V. Lunev <den@openvz.org>
Signed-off-by: Alexey Dobriyan <adobriyan@openvz.org>
Signed-off-by: David S. Miller <davem@davemloft.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 net/ipv6/ndisc.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/net/ipv6/ndisc.c
+++ b/net/ipv6/ndisc.c
@@ -736,7 +736,7 @@ static void ndisc_recv_ns(struct sk_buff
 				 * so fail our DAD process
 				 */
 				addrconf_dad_failure(ifp);
-				goto out;
+				return;
 			} else {
 				/*
 				 * This is not a dad solicitation.
